syntax = "proto3";

package instruction;

option java_package = "edu.cmu.cs.gabriel.instruction";
option java_outer_classname = "Protos";

message Sandwich {
    enum State {
        START = 0;
        NOTHING = 1;
        BREAD = 2;
        HAM = 3;
        LETTUCE = 4;
        CUCUMBER = 5;
        HALF = 6;
        TOMATO = 7;
        HAM_WRONG = 8;
        FULL = 9;
    }

    State state = 1;
    double holo_x = 2;
    double holo_y = 3;
    double holo_depth = 4;
}

message EngineFields {
    int64 update_count = 1;
    oneof state {
        Sandwich sandwich = 3;
    }
}
